### PR TITLE
PF-2852: Make the test more resilient to failure

### PR DESCRIPTION
**Pull request by yuhuyoyo**
one of the test failed during update assertion so we never reached the end where we reset the bucket name. 

In the failure, the last updated timestamp is off. My guess is that in that the hook where we log the update is slightly late so we are using the previous timestamp in the updateResuorce and got the newer timestamp in the gotResource, causing the disparity.

I removed the comparison. just asserting the updated resource got the new name, new description, etc should be enough.

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java [hunk 1]
```diff
@@ -138,14 +155,8 @@ public void update() throws Exception {
...
             newCloningInstruction);
-    // Update the sourceResource to the updated one as all the tests are sharing
-    // the same resource.
-    ApiGcpGcsBucketResource getResource =
-        mockMvcUtils.getReferencedGcsBucket(
-            userAccessUtils.defaultUserAuthRequest(),
-            workspaceId,
-            sourceResource.getMetadata().getResourceId());
-    assertEquals(updatedResource, getResource);
+    bucketIsUpdated = true;
```
Comment (zloery) [inline service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java:158]: nit: this should be set right before the `updateReferencedGcsBucket` call instead to prevent the tiny edge case where the call causes a test failure but still makes the update happen

File: service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java [hunk 2]
```diff
@@ -191,15 +194,10 @@ public void update_nameAndDescriptionOnly() throws Exception {
             newDescription,
             /*newBucketName=*/ null,
             newCloningInstruction);
+    bucketIsUpdated = true;
```
Comment (zloery) [inline service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java:197]: same here

**Review by zloery — APPROVED** [review]

File: service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java [hunk 3]
```diff
@@ -92,6 +95,20 @@ public void cleanup() throws Exception {
...
+    if (bucketIsUpdated) {
+      mockMvcUtils.updateReferencedGcsBucket(
+          userAccessUtils.defaultUserAuthRequest(),
+          workspaceId,
+          sourceResource.getMetadata().getResourceId(),
+          sourceResourceName,
+          RESOURCE_DESCRIPTION,
+          sourceBucketName,
+          ApiCloningInstructionsEnum.NOTHING);
+    }
```
Comment (ddietterich) [inline service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java:109]: I think this should restore `bucketIsUpdated` to `false`, yes?

**Review by ddietterich — COMMENTED** [review]
IIUC, the `@AfterEach` is to restore the reference resource to an initial state.

I this it would be more straightforward to create a new reference resource for each test, so that there would not be interaction between the tests.

File: service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java (hunk 3 above)
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/app/controller/ReferencedGcpResourceControllerGcsBucketConnectedTest.java:110]: good catch

**Comment by yuhuyoyo** [comment]
> IIUC, the `@AfterEach` is to restore the reference resource to an initial state.
> ya.
> I this it would be more straightforward to create a new reference resource for each test, so that there would not be interaction between the tests.

just slightly faster? there's only two update tests.

**Comment by ddietterich** [comment]
>  just slightly faster? there's only two update tests.

I would suggest making the reference in each test.

**Comment by yuhuyoyo** [comment]
> > just slightly faster? there's only two update tests.
> 
> I would suggest making the reference in each test.

alright

**Review by ddietterich — APPROVED** [review]
